documentation/cadl-structure-guidelines.md

```diff
@@ -0,0 +1,127 @@
+      -> Compute.Disk.Management          (service)
+      -> Compute.Gallery.Management       (service)
+      -> Compute.Skus.Management          (service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+   -> keyvault
+      -> KeyVault.Certificates            (data-plane SDK + service)
+      -> KeyVault.Keys                    (data-plane SDK + service)
+      -> KeyVault.Secrets                 (data-plane SDK + service)
+      -> KeyVault.Management              (mgmt SDK + service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+```
+
+## Service Folders
+
+The service folder contains the entire Cadl library specification for a service, which could include custom linter rules, methods, etc.
+
+The folder name should correspond to the RP-name, but dropping the leading "Azure" or "Microsoft" prefix. Two additional naming conventions should be following:
+
+- `Management` at the end of the service RP-name indicates a management (resource manager) libarary.
+- `Shared` at the end of the name indicates a shared library. If a management library has a shared component (unlikely), `Shared` should follow `Management`.
+
+### Packages
+
+All services and service family libraries are modeled as Cadl packages, since you must install supporting librarie for proper tooling support. Per this proposal, all packages defined in the repo would be **unpublished**. Only packages in `cadl-azure` would be published. We will annotate the packages with `"private": true` in the `package.json` file to prevent publishing these to npm.
+
+All packages defined in this repo would use the `@cadl-api-spec` scope and use a lowercased, dashed form of the service namespace (ex: `@cadl-api-spec/azure-storage-blob`).
+
+### Structure
+
+Each package should have the following minimum structure:
+
+- `cadl-project.yaml`
+- `package.json`
+- `main.cadl` file
+- Supporting `*.cadl` files
+- `examples/` folder for example JSON files
+
+Authors may use folders as desired for organizing cadl files.
+
+Additionally, packages which wish to define custom linter rules or otherwise use TypeScript must place any `*.ts` files under a `src/` folder. Within the `src/` folder, the author may use as many subfolders as desired for organizing code.
+
+To distinguish between folders which define a service, an SDK, or both, one can look to the `cadl-project.yaml` and/or the `package.json`.
+
+- SDKs will take dependencies on the `@azure-tools/cadl-dpg` library, as well as SDK-specific emitters such as `@azure-tools/cadl-python` and configure them within `cadl-project.yaml`.
+- SDKs _may_ have a sidecar file to customize how the SDK is shaped. Folders that describe service definitions only will not have a sidecar file. _Note: the absence of a sidecar does not mean that a folder does not describe an SDK, but the presence of one means it is an SDK._
```

Comment:
Where is the sidecar file? is it in the folder together with `cadl-project.yaml`? Or it's in sdk repository?

Reply:
It should live in the same folder as the cadl-project.yaml file. However as @tjprescott calls out here there doesn't necessarily always need to be a sidecar file and there also might only be a single one as opposed to one per-language. Those details are still be worked out.